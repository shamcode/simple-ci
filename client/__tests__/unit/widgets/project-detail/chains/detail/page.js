import { DI } from 'sham-ui';
import Page from '../../../../../../src/widgets/project-detail/chains/detail/page';
import renderer from 'sham-ui-test-helpers';


afterEach( () => {
    DI.bind( 'store', null );
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    DI.bind( 'store', {
        getProjectChainById: jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) ),
        getProjectById: jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) )
    } );
    DI.bind( 'router', {
        lastRouteResolved: jest.fn().mockReturnValueOnce( { params: { id: 1 } } ),
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );
    const meta = renderer( Page, {} );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
