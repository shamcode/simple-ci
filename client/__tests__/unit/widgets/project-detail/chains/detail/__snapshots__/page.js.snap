// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {},
  "html": "
<!--if-->
<!--FlushPageDataLoading-->
<!--if-->
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "afterRender": [Function],
    "chain": Object {},
    "dataLoaded": false,
    "dataSaving": false,
    "errors": Array [],
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
